src/cpu/aarch64/jit_sve_512_x8s8s32x_conv_kernel.hpp

```diff
@@ -0,0 +1,296 @@
+/*******************************************************************************
+* Copyright 2016-2020 Intel Corporation
+* Copyright 2020 FUJITSU LIMITED
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef CPU_AARCH64_JIT_SVE_512_X8S8S32X_CONV_KERNEL_HPP
+#define CPU_AARCH64_JIT_SVE_512_X8S8S32X_CONV_KERNEL_HPP
+
+#include "common/c_types_map.hpp"
+#include "common/memory_tracking.hpp"
+
+#include "cpu/aarch64/jit_generator.hpp"
+#include "cpu/aarch64/jit_primitive_conf.hpp"
+
+#define ADDMAX 4095
+#define MOVMAX 65535
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace aarch64 {
+
+using namespace Xbyak_aarch64;
+
+template <typename Vmm>
+struct _jit_sve_512_x8s8s32x_fwd_kernel : public jit_generator {
+    DECLARE_CPU_JIT_AUX_FUNCTIONS(_jit_sve_512_x8s8s32x_conv_fwd_ker_t)
+
+    enum { STATE_FIRST_DST_LOAD = 0x1U };
+
+    _jit_sve_512_x8s8s32x_fwd_kernel(
+            const jit_conv_conf_t &ajcp, const primitive_attr_t &attr)
+        : jcp(ajcp), attr_(attr) {
+        if (jcp.with_eltwise) assert(!"not supported");
+    }
+
+    ~_jit_sve_512_x8s8s32x_fwd_kernel() {}
+
+    jit_conv_conf_t jcp;
+    const primitive_attr_t &attr_;
+
+private:
+    const int ic_sub_step = 4;
+
+    enum {
+        typesize = sizeof(float),
+        ker_reg_base_idx = 28,
+        ker_dw_reg_base_idx = 30,
+        ker_zp_reg_base_idx = 26,
+    };
+    typedef enum {
+        no_last_block,
+        last_ic_block,
+        last_sp_block,
+    } ic_block_t;
+
+    /* data regs */
+    const XReg reg_ptr_scales = x7; //rax;
```

Comment:
nit: If ARM does not have alternate bynames for registers (e.g. `rax`, `rbx`, etc.) then guess it should be removed from the comments.

Reply:
Thx! I'll delete it.